raiden-ts/src/services/epics/udc.ts

```diff
@@ -324,54 +343,40 @@ export function udcCheckWithdrawPlannedEpic(
  * @param deps.signer - Signer instance
  * @param deps.provider - Provider instance
  * @param deps.config$ - Config observable
- * @returns Observable of udcWithdraw.success|udcWithdrawPlan.failure actions
+ * @returns Observable of udcWithdraw.success|udcWithdraw.failure actions
  */
-export function udcWithdrawPlannedEpic(
+export function udcWithdrawEpic(
   action$: Observable<RaidenAction>,
-  state$: Observable<RaidenState>,
+  {}: Observable<RaidenState>,
   { log, userDepositContract, address, signer, provider, config$ }: RaidenEpicDeps,
-): Observable<udcWithdraw.success | udcWithdrawPlan.failure> {
+): Observable<udcWithdraw.success | udcWithdraw.failure> {
   return action$.pipe(
-    filter(udcWithdrawPlan.success.is),
-    filter((action) => action.payload.confirmed === true),
-    mergeMap((action) =>
-      state$.pipe(
-        pluck('blockNumber'),
-        filter((blockNumber) => action.payload.block < blockNumber),
-        take(1),
-        mapTo(action),
-      ),
-    ),
-    mergeMap((action) =>
-      retryAsync$(
-        () =>
-          userDepositContract.callStatic
-            .balances(address)
-            .then((balance) => [action, balance] as const),
-        provider.pollingInterval,
-        { onErrors: networkErrors },
-      ),
-    ),
-    concatMap(([action, balance]) => {
+    filter(udcWithdraw.request.is),
+    concatMap((action) => {
       const contract = getContractWithSigner(userDepositContract, signer);
-      return defer(() => {
-        assert(balance.gt(Zero), [
-          ErrorCodes.UDC_WITHDRAW_NO_BALANCE,
-          {
-            balance: balance.toString(),
-          },
-        ]);
-        return contract.withdraw(action.meta.amount);
-      }).pipe(
+      let balance: UInt<32>;
+      return retryAsync$(
+        async () => userDepositContract.callStatic.balances(address),
+        intervalFromConfig(config$),
+        { onErrors: networkErrors, log: log.info },
+      ).pipe(
+        mergeMap(async (balance_) => {
```

Comment:
This appears to be a more often used pattern. Is it worth to export this? 🤔

Reply:
Will not be handled.

Reply:
Thanks for the headsup.. actually, the first `retryAsync$` is not needed since it's also covered by the `retryWhile` there, so I'll simplify this code a little bit more. About the pattern, it can be not very practical to split, since this references the context variable (`balance`), and also usually have specific error messages and contexts which would be more verbose to pass all in a custom operator call than to do it like that. But we can re-evaluate this if the pattern spreads more.

Reply:
:heavy_check_mark: